### PR TITLE
Code Review - JS

### PR DESCRIPTION
**Pull request by SRDewan**

### CODE REVIEW

File: experiment/quiz.json [hunk 1]
```diff
@@ -0,0 +1,258 @@
+{
+    "artciles": [
```
Comment (raj-vlabs) [inline experiment/quiz.json:2]: Misspelling

File: experiment/references.md [hunk 2]
```diff
@@ -1 +1,3 @@
-### Link your references in here
\ No newline at end of file
+- IS : 2720 (Part II) – 1973, Method of Test for soil : Part II
+- Soil Mechanics and Foundations.
```
Comment (raj-vlabs) [inline experiment/references.md:2]: If this is a book, please give the names of authors, publisher etc

File: experiment/simulation/css/main.css [hunk 3]
```diff
@@ -1 +1,55 @@
...
+	order: 2;
+}
+
+.observations {
+	order: 3;
+}
+
+@media screen and (max-width: 1023px) {
+	.workspace {
+		flex-wrap: wrap;
```
Comment (raj-vlabs) [inline experiment/simulation/css/main.css:41]: Did you try a variant of Bulma's is-flex-wrap-wrap for this property? Also see if is-multiline works

File: experiment/simulation/css/main.css [hunk 4]
```diff
@@ -1 +1,55 @@
...
+	}
+
+	.apparatus {
+		order: 1;
```
Comment (raj-vlabs) [inline experiment/simulation/css/main.css:45]: It's interesting that Bulma doesn't offer an order class whereas bootstrap does.

File: experiment/simulation/index.html [hunk 5]
```diff
@@ -1,13 +1,72 @@
...
+	<head>
+		<meta charset="utf-8">
+		<meta name="viewport" content="width=device-width, initial-scale=1">
+		<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bulma@0.9.1/css/bulma.min.css">
+		<link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/virtual-labs/virtual-style@0.0.5-b/css/style.min.css">
+		<link rel="stylesheet" href="./css/main.css">
+	</head>
+
+	<body class="page">
+		<div class="columns is-centered is-flex">
```
Comment (raj-vlabs) [inline experiment/simulation/index.html:12]: Why do you need is-flex here? Is that even a valid class?
columns is flex by default

**Review by raj-vlabs — CHANGES_REQUESTED** [review]
1. Is the structure of quiz.json the new quiz structure that we are going to follow?
2. I am not able to see pretest and posttest. Are they both going to be built from quiz.json?
3. From what I understood, I like the code. It's good to encapsulate different components into their own classes. It moves us in the direction of converting them into independent web components.

File: experiment/simulation/css/main.css (hunk 4 above)
Comment (SRDewan) [reply experiment/simulation/css/main.css:45]: Ya, I searched but could not find anything in Bulma for reordering.

File: experiment/simulation/index.html (hunk 5 above)
Comment (SRDewan) [reply experiment/simulation/index.html:12]: Yes, it is a valid class and although columns is flex by default, for some reason it gets overridden for the main workspace and ends up messing up the config for smaller screens.

**Comment by SRDewan** [comment]
>     1. Is the structure of quiz.json the new quiz structure that we are going to follow?
> 
>     2. I am not able to see pretest and posttest. Are they both going to be built from quiz.json?
> 
>     3. From what I understood, I like the code. It's good to encapsulate different components into their own classes. It moves us in the direction of converting them into independent web components.

Yes, so I haven't worked on the quizzes yet. I will replace the quiz.json file with pretest.json and posttest.json files when I get it to it with the SME.

**Comment by SRDewan** [comment]
Done with everything now, including quizzes. So now this experiment can be reviewed completely and merged once done. @raj-vlabs

File: experiment/simulation/js/main.js [hunk 6]
```diff
@@ -1 +1,531 @@
...
+		};
+	};
+
+	class weight{
+		constructor(height, width, x, y) {
+			this.height = height;
+			this.width = width;
+			this.pos = [x, y];
+			this.img = new Image();
+			this.img.src = './images/weighing machine.png';
```
Comment (raj-vlabs) [inline experiment/simulation/js/main.js:167]: Try to not use spaces in file names, instead use hyphens

**Review by raj-vlabs — COMMENTED** [review]
The code looks good. There are very few explanations for wrong options in posttest. Are we waiting for the SME or this is going to stay this way?

**Comment by SRDewan** [comment]
> The code looks good. There are very few explanations for wrong options in posttest. Are we waiting for the SME or this is going to stay this way?

It is mostly going to stay this way. SME didn't provide many explanations. But I can try adding some on my own?

**Review by raj-vlabs — APPROVED** [review]

**Pull request merged**